### DISCUSSION TITLE
Variables exercise - awk

### DISCUSSION THREAD

**Discussion by eugeniadamato** [sandbox-bio/sandbox.bio#22] [Q&A] [answered]
I found a solution for the exercise in page #8, the following code gives the result =15 
 
awk -F "\t" ' { if($3 = "Bottled Water") print $2}'  orders.tsv > temp1 | sort -nr temp1 | awk -F  "\t" '{ if(NR == 1) print $1 }' temp1 > largest_order.tsv

but the website does not pick it up as resolved.
There might be more elegant results (I am a beginner). 

If someone obtained an approved result can you share it please ?
thanks in advance

**Comment by robertaboukhalil** [comment, marked as answer]
Hi @eugeniadamato, 

A few things to look into:

1. Take a closer look at your awk statement `if($3 = "Bottled Water")`. It's supposed to only keep rows where the 3rd column is Bottled Water. BUT if you calculate the number of lines after filtering, you'll see it's exactly the same as without filtering, so something is off:
```bash
guest@sandbox$ wc -l orders.tsv 
4623 orders.tsv

guest@sandbox$ awk -F "\t" ' { if($3 = "Bottled Water") print $2}' orders.tsv | wc -l
4623
```

3. When doing `orders.tsv > temp1 | sort`, the issue is you shouldn't mix both patterns of `command > file` and `command | command`. You can either rewrite this so there's no `> temp1` and you keep piping the output of `awk` to `sort` directly without mentioning `temp1`. Or you can rewrite it as 2 commands, where you store the output of `awk` into `temp1`, then that's the end of the first command. And separately you run `sort -nr temp1 | awk -F "\t" '{ if(NR == 1) print $1 }' ` (notice how you don't need to specify `temp1` in the awk command because awk is reading from the piped data.

Hope this helps!

**Reply by eugeniadamato** [reply]
thank you so much for your reply.
I made this new code:
awk -F "\t" ' { if($3 = "Bottled Water") print$2 }'  orders.tsv | sort -nr | awk -F "\t" '{ if(NR == 1) print $1 }' > largest_order.tsv

"largest_order.tsv" has only one number,  15, but the site still doesn't pick it up as correct.
Maybe I misunderstood the question. Exploring the file:

order_id quantity Bottled Water choice_description item_price
1 1 Bottled Water NULL $2.39 
1 1 Bottled Water [Clementine] $3.39 
1 1 Bottled Water [Apple] $3.39 
1 1 Bottled Water NULL $2.39 
2 2 Bottled Water [Tomatillo-Red Chili Salsa (Hot), [Black Beans, Rice, Cheese, Sour Cream]] $16.98 
3 1 Bottled Water [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sour Cream, Guacamole, Lettuce]] $10.98 
3 1 Bottled Water NULL $1.69 
4 1 Bottled Water [Tomatillo Red Chili Salsa, [Fajita Vegetables, Black Beans, Pinto Beans, Cheese, Sour Cream, Guacamole, Lettuce]] $11.75 
4 1 Bottled Water [Tomatillo Green Chili Salsa, [Pinto Beans, Cheese, Sour Cream, Lettuce]] $9.25 

I see many rows for the same order_id. Is the question how many water bottles, whatever the type, were consumed in total at the same table ?

**Reply by robertaboukhalil** [reply]
Almost there! The only thing wrong is `if($3 = "Bottled Water")`. You're setting column 3 to the value `Bottled Water`, but you should be comparing the values using `==`.

**Comment by deve105** [comment]
it works for me with:
awk -F "\t" ' { if($3 =="Bottled Water") print $2}' orders.tsv | sort -nr | head -1 > largest_order.tsv
